pkg/service/targetgroupsvc/update.go

```diff
@@ -0,0 +1,58 @@
+package targetgroupsvc
+
+import (
+	"context"
+	"strings"
+
+	"github.com/common-fate/common-fate/pkg/storage"
+	"github.com/common-fate/common-fate/pkg/targetgroup"
+	"github.com/common-fate/common-fate/pkg/types"
+	"github.com/common-fate/ddb"
+	"go.uber.org/zap"
+)
+
+type UpdateOpts struct {
+	TargetGroupID string
+	UpdateRequest types.CreateTargetGroupRequest
+}
+
+func (s *Service) UpdateTargetGroup(ctx context.Context, req UpdateOpts) (*targetgroup.TargetGroup, error) {
```

Comment:
Based on the linear ticket https://linear.app/common-fate/issue/CF-624/create-api-methods-for-targetgroup-and-deployment-as-per-spec

The update method is not a requirement for this PR, you should exclude it as the behaviour is not define in our designs